lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -396,6 +403,29 @@ public function getMaxResults()
         return $this->maxResults;
     }
 
+    /**
+     * Sets the flag to only retrieve distinct results
+     *
+     * @return $this This QueryBuilder instance.
+     */
+    public function distinct()
```

Comment:
You should allow users reset the flag back to `false`. Please add `$flag = true` argument to achieve this.